testdata/project-v4-with-plugins/internal/controller/memcached_controller.go

```diff
@@ -117,37 +118,22 @@ func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
-	// Let's add a finalizer. Then, we can define some operations which should
-	// occur before the custom resource is deleted.
-	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers
-	if !controllerutil.ContainsFinalizer(memcached, memcachedFinalizer) {
```

Comment:
@mateusoliveira43 Why remove the func from controller-runtime used to add the finalizer we should keep it as it is

Reply:
this is before logic to delete resource  https://github.com/kubernetes-sigs/kubebuilder/blob/cbc6e383c342f1337ab37ee4aa0755957a01f9c7/testdata/project-v4-with-plugins/internal/controller/memcached_controller.go#L139

I was concerned if this fails https://github.com/kubernetes-sigs/kubebuilder/blob/cbc6e383c342f1337ab37ee4aa0755957a01f9c7/testdata/project-v4-with-plugins/internal/controller/memcached_controller.go#L185

we would add finalizer again and try to delete again. I think this is corner case, so should be ok to let as is